daemon/api_debug.go

```diff
@@ -128,7 +134,58 @@ func getChangeTimings(st *state.State, changeID string) Response {
 			UndoingTimings: undoingTimingsByTask[t.ID()],
 		}
 	}
-	return SyncResponse(m, nil)
+	return m, nil
+}
+
+func getChangeTimings(st *state.State, changeID, ensureTag string, allEnsures bool) Response {
+	// If ensure tag was passed by the client, find its related changes;
+	// we can have many ensure executions and their changes in the responseData array.
+	if ensureTag != "" {
+		ensures, err := timings.Get(st, -1, func(tags map[string]string) bool {
```

Comment:
Nitpick: this could be captured in `func collectEnsureTimings(st *state.State, ensureTag string) ([]*debugTimings, error)` helper